### ISSUE TITLE
has_instance and has_instance2 always return  Failure

### ISSUE THREAD

**Issue by LingLingYang** [HerculesWS/Hercules#1397]
```
prontera,154,173,5  script  test321 PORING,{

    if(has_instance("prontera")=="") {
        set .@id,instance_create("prtdup02",0,IOT_NONE);
        instance_attachmap("prontera", .@id,1,"prontera02");    
        instance_init .@id;
    }
    mes "instance has been created!";
    next;
    mes "instance map name is:"+has_instance("prontera");
    mes "instance id is:"+has_instance2("prontera");

    close;

}
```

<img width="150" alt="1222" src="https://cloud.githubusercontent.com/assets/19987693/17578682/9c1d93a2-5fbe-11e6-9ad7-bd2fe17293a9.png">

**Comment by Aeromesi** [comment]
Incorrect. It's just your instance script isn't right. If you need help on instance scripting I'm pretty good at it! Just message me on Hercules or view My Profile for my Skype info. But here's a redefined code that deletes the instance if you click the NPC again and it shows the map name as well as the id being >= 0 which is located in the script_commands.txt "It will return -1 upon failure, valid instance ids are >= 0."

It also primarily had to do with IOT_NONE, for some reason IOT_NONE is just an open instance that can be generated that isn't attached to any character? IOT_CHAR brought me the return values as positive, here's a screenshot.

![image](https://cloud.githubusercontent.com/assets/6835123/17583632/b5ef773a-5f66-11e6-9acd-66ce80e12bc8.png)

When I do it with IOT_NONE on the other hand does not. Let me try generating an instance through OnInit via IOT_NONE and see if my theory is correct

**Comment by Aeromesi** [comment]
Oh and the code:

`prontera,154,173,5	script	test321	PORING,{
if ( !.loaded ) {
	.@md_name$ = "Value Test";
	.@instance = instance_create( .@md_name$, getcharid( 3 ), IOT_CHAR);
	if( .@instance < 0 ) { 
		dispbottom "Instance Failed.";
		end;
	}
	dispbottom "^0000ff" + .@md_name$ + "^000000- Attempting to create Instance.";
	if( instance_attachmap("prontera",.@instance) == "" ) {
		instance_destroy( .@instance );
		dispbottom "Failed to instance prontera/Instance Name: prontera";
		end;
	}
	instance_set_timeout( 7200, 300, .@instance ); // 3 Hours to complete
	instance_init( .@instance );
	.loaded++;
	mes "name: "+has_instance("prontera")+".";
	mes "id: "+has_instance2("prontera")+".";
close;

} else
instance_destroy ( .@instance );
mes "instance deleted";
.loaded = 0;
close;
}`

**Comment by Aeromesi** [comment]
And I just figured out that it's not possible to create instances without at least IOT_NONE which defaults to party so I was wrong, but anyways IOT_CHAR seems to fix your issue... xD

**Comment by LingLingYang** [comment]
Thanks for your explanation, but dunno why has_instance and has_instance2 should attach rid , party or guild. I wonder would it be possible these two script commands also work with IOT_NONE instance ?
just a suggestion and hope devs can make them more convenient :)

**Comment by Smokexyz** [comment]
Why would you need to create an instance without a character, guild or party to use it? Ideally you'd want player to enter the instance and therefore revolve instance data around the object that has created it.

**Comment by iRagno** [comment]
@Smokexyz: It is useful to instance things that aren't attached to anyone, such as battlegrounds, or maybe events that requires participation from anyone and needs to be not attached to somebody specific.

**Comment by MishimaHaruna** [comment]
This should be fixed as of #1469
